ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/PartyManagementServiceIT.scala

```diff
@@ -93,6 +94,32 @@ final class PartyManagementServiceIT extends LedgerTestSuite {
     }
   })
 
+  test(
+    "PMRejectionDuplicateHint",
```

Comment:
@rgugliel-da can you confirm that this is indeed how canton reacts to duplicate parties

Reply:
Since `ledger-api-test-tool-on-canton` exercises this, this is the current Canton behavior afaict. But good to make sure if it is indeed intended

Reply:
Yep, that's correct.

Reply:
@tudor-da If I understand correctly:
- the participant now _must_ reject an allocation if the same hint was used before
- it is undefined how the hint affects the allocated party name

So even ledgers that completely discard the hint when generating party names must reject duplicate hints?  If yes, then we should update the protobuf comments in [party_management_service](https://github.com/digital-asset/daml/blob/main/ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/party_management_service.proto#L59) accordingly. Also, remove the references to `daml-on-sql` and `daml-on-kv-ledger` and replace them with more generic text.